src/vpu/encoder_h264.c

```diff
@@ -191,6 +193,21 @@ gboolean gst_imx_vpu_encoder_h264_set_open_params(GstImxVpuEncoderBase *vpu_enco
 {
 	GstCaps *template_caps, *allowed_caps;
 	GstImxVpuEncoderH264 *vpu_encoder_h264 = GST_IMX_VPU_ENCODER_H264(vpu_encoder_base);
+	GstVideoFormat fmt = GST_VIDEO_INFO_FORMAT(&(input_state->info));
+
+	if (fmt == GST_VIDEO_FORMAT_GRAY8)
+	{
+		open_params->color_format = IMX_VPU_COLOR_FORMAT_YUV400;
```

Comment:
On Fri, Sep 09, 2016 at 05:03:00AM -0700, dv1 wrote:
> > @@ -191,6 +193,21 @@ gboolean gst_imx_vpu_encoder_h264_set_open_params(GstImxVpuEncoderBase *vpu_enco
> >  {
> >  	GstCaps *template_caps, *allowed_caps;
> >  	GstImxVpuEncoderH264 *vpu_encoder_h264 = GST_IMX_VPU_ENCODER_H264(vpu_encoder_base);
> > +	GstVideoFormat fmt = GST_VIDEO_INFO_FORMAT(&(input_state->info));
> > +
> > +	if (fmt == GST_VIDEO_FORMAT_GRAY8)
> > +	{
> > +		open_params->color_format = IMX_VPU_COLOR_FORMAT_YUV400;
> 
> Hmm, this works? IMX_VPU_COLOR_FORMAT_YUV400 isn't actually accepted by imxvpuapi for anything other than for MJPEG. This might also not be necessary - your nice trick with the dummy plane should also work with YUV420.

It doen't seem to hurt, but you're right : it is not necessary.